### DISCUSSION TITLE
can someone please help me with understanding the structure of code

### DISCUSSION THREAD

**Discussion by sharma-tanishq** [excalidraw/excalidraw#5340] [Q&A] [answered]
can someone please help me with understanding the structure of code

**Comment by ad1992** [comment, marked as answer]
Hi @sharma-tanishq 👋🏻, is there anything specific you are looking for ?

In any case here is a general overview which will help you get started, Excalidraw App internally uses the excalidraw npm package, docs here - https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README_NEXT.md
and adds more functionalities eg collaboration on top of it. 
So you can start looking into this file https://github.com/excalidraw/excalidraw/blob/master/src/excalidraw-app/index.tsx as an entry point to understand the flow and then navigate accordingly.

**Reply by sharma-tanishq** [reply]
Thank you, I just want to understand how things are working in this suppose if i want to add a second dark mode button in the side of toolbar(the one with the pencil icon) how will i do it? There is the darkModeToggle file but how should it be used? . can you please help me figure this out

**Reply by ad1992** [reply]
Are you forking the repo or using the npm package ? In case you are using the npm package you can use `render` props in the package https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README_NEXT.md#renderTopRightUI, https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README_NEXT.md#renderFooter but this will add the button either to top right or as footer as show in the [example](https://excalidraw-package-example.vercel.app/)

**Reply by sharma-tanishq** [reply]
Thank you so much, I can now atleast insert components on the screen

**Reply by ad1992** [reply]
Awesome 👍🏻

**Reply by DebakantPradhan** [reply]
I am forking the repo
